PBTA Histologies: Integrated molecular subtyping to base histology (7 of N) (#870)
AlexsLemonade/OpenPBTA-analysis: analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd

  left_join(compiled_subtyping,by=c("Kids_First_Biospecimen_ID","sample_id","Kids_First_Participant_ID","tumor_descriptor")) %>%
  filter(toupper(broad_histology.x) != toupper(broad_histology.y)) %>%
  select(Kids_First_Biospecimen_ID,pathology_diagnosis,pathology_free_text_diagnosis,starts_with("broad_histology"),starts_with("short_histology")) %>%
  unique()

diff_broad_histology
```

#### Check differences in short_histology
Here we want to check for short_histology changes not part of `diff_broad_histology`.
This will help us check what string assignment has changed from `molecular-subtyping-pathology`

```{r}

diff_short_histology <- base_histology %>% 
  filter(!Kids_First_Biospecimen_ID %in% diff_broad_histology$Kids_First_Biospecimen_ID) %>%
  select(-Notes,-molecular_subtype) %>%
  left_join(compiled_subtyping,by=c("Kids_First_Biospecimen_ID","sample_id","Kids_First_Participant_ID","tumor_descriptor")) %>%
  filter(toupper(short_histology.x) != toupper(short_histology.y) 
         ) %>%
  select(Kids_First_Biospecimen_ID,pathology_diagnosis,pathology_free_text_diagnosis,starts_with("broad_histology"),starts_with("short_histology")) %>%
  unique()

diff_short_histology
```

> The above 172 changes occured because of changes in string value assignement
> In compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv sample with broad_histology== `Ependymal tumor` is `EPN`	but in base histology it is `Ependymoma`.
> For samples with broad_histology==`Embryonal tumor`, short_histology is `Embryonal tumor` but in base histlogy it was `ETMR`

Reviewer: ```suggestion
The above 172 changes occurred because of changes in string value assignment
In `compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv` sample with broad_histology== `Ependymal tumor` is `EPN`, but in base histology it is `Ependymoma`.
For samples with broad_histology==`Embryonal tumor`, short_histology is `Embryonal tumor` but in base histology it was `ETMR`.
```

---
title: "Integrate molecular subtyping results"
output: 
  html_notebook:
    toc: true
    toc_float: true
author: Krutika Gaonkar for D3b
date: 2020
params:
   is_ci: FALSE
---

The purpose of this notebook is to integrate molecular subtyping results from 
[molecular-subtyping-pathology](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/molecular-subtyping-pathology) with `pbta-histologies-base.tsv`.

> Here we are using v18 `pbta-histologies-base.tsv`

Reviewer 2: Just to clarify, (and sorry if this is written elsewhere, but it may be good to add it here too) is the purpose of `pbta-histologies-base.tsv` that you write the subtyping results there and that in the next version of the release, the results in `pbta-histologies-base.tsv` are added to `pbta-histologies.tsv`?

Reviewer 2: Aka it seems like `pbta-histologies-base.tsv` is like the working draft?

Author: Right, we can call `pbta-histologies-base.tsv` a working draft. 
In each release there will be a `pbta-histologies-base.tsv` are added to `pbta-histologies.tsv`. The **PBTA Histologies 1-7** PR  use the base histology file which have any integrated_diagnosis,Notes and molecular_subtype as NA through all the molecular-subtyping-XYZ . Through this process we gradually create the `pbta-histologies.tsv` for the same release.

You might have noticed that the modules (other than subtyping) were all using RNA data.
https://github.com/kgaonkar6/OpenPBTA-analysis/blob/760f7a42d253edcaf3341ac0873634e55590e59e/scripts/run-for-subtyping.sh#L13-L32

We only ran these modules because along with the `pbta-histologies-base.tsv` we also needed to update these files with the new RNA samples to use in our molecular subtyping modules.

Reviewer 2: Can you put some form of your description here in the beginning of this notebook?

## Set up

```{r}
library(tidyverse)
data_dir <- "../../data/release-v18-20201123/"
base_histology <- read_tsv(file.path(data_dir,"pbta-histologies-base.tsv"),
                           col_types = readr::cols(molecular_subtype = readr::col_character(),
                                                   short_histology = readr::col_character(),
                           broad_histology = readr::col_character(),
                           Notes = readr::col_character())) %>%
  unique()
```

### Read molecular-subtyping-pathology results

> Reading molecular_subtype,	integrated_diagnosis, short_histology,	broad_histology and	Notes from `compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv` 

Reviewer 2: This spacing is odd. Is there a reason behind this odd spacing?

Reviewer 2: If you want to do a quick thing to try to get rid of some of the weird spacing in this notebook, you can run `styler::style_file("analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd")` and see if styler can fix some of these things for you. Beyond that, I don't know that we need to fuss with it. 
...
I just checked, and unfortunately we do not have the `styler` package in the docker container. So don't worry about this, but it could be handy for you elsewhere.

Author: hmm not sure where these spaces came from , checking

Author: the spaces looked fine on localhost, I edited it and looks ok here as well. Let me know if I missed any weird spaces

```{r}

> compiled_subtyping<-read_tsv("../molecular-subtyping-pathology/results/compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv")

Reviewer 2: ```suggestion
compiled_subtyping<-read_tsv(file.path("..", "molecular-subtyping-pathology", "results", "compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv")
```
You may want to alter the spacing on my suggestion depending on how it looks.

```

Update "Other" sample broad/short histology and harmonized_diagnosis and add to `compiled_subtyping`

```{r}

> Other_subtypes <- read_tsv(file.path("input","pathology_dx and pathology_free_text_diagnosis to broad_histology for subtyping module - rules_without_subtype_WIP.tsv"))

Reviewer 2: I'm not a fan of this file name but I'm guessing it comes from somewhere else and we don't have control over that.

Author: umm I could update the name and add the link to the googlesheet tab used, just wanted to keep the name as is in the google sheet for reference.

Reviewer 2: I think the file name should be changed everywhere or not at all. But if changing it will cause confusion, don't bother, no worries.

    short_histology = if_else(!is.na(short_histology.subtyped),
                              short_histology.subtyped,
                              short_histology.base),
    harmonized_diagnosis = 
      case_when(!is.na(integrated_diagnosis) ~ integrated_diagnosis,
                is.na(integrated_diagnosis) & 
                  !is.na(harmonized_diagnosis) ~ harmonized_diagnosis,
                is.na(integrated_diagnosis) & 
                  is.na(harmonized_diagnosis) &
                  !is.na(pathology_diagnosis) ~ pathology_diagnosis
    )) 
```

### Check if any duplicates

```{r}
dup_ids<-histology$Kids_First_Biospecimen_ID[duplicated(histology$Kids_First_Biospecimen_ID)]

histology[which(histology$Kids_First_Biospecimen_ID %in% dup_ids),]
```

No duplicates

### Check if broad_histology, short_histology or harmonized_diagnosis

Are there NA in broad_histology, short_histology or harmonized_diagnosis

```{r}
histology %>% 
  filter(sample_type=="Tumor",
         (is.na(broad_histology)| is.na(short_histology)| is.na(harmonized_diagnosis))) %>%
  tally()

```

No NAs in broad_histology, short_histology or harmonized_diagnosis

> Just a note, integrated_diagnosis == NA is expected for samples where subtyping is not performed or is molecular_subtype == "XYZ,To be classified" because no evidence was provided/available for these samples we are not able to add integrated_diagnosis.

Reviewer 2: ```suggestion
Just a note, integrated_diagnosis is expected to be `NA` for samples where subtyping is not performed or if molecular_subtype is "XYZ,To be classified". 
This means no evidence was provided/available for these samples so we are not able to add integrated_diagnosis.
```

#### Check differences in broad_histology 
Checking for differences in broad_histology to look for changes in molecular_subtype 

```{r}
diff_broad_histology<- histology %>%
  filter(toupper(broad_histology.base) != toupper(broad_histology.subtyped)) %>%
  select(Kids_First_Biospecimen_ID,starts_with("broad_histology"),starts_with("short_histology")) %>%
  unique()

diff_broad_histology
```

#### Check differences in short_histology
Here we want to check for short_histology changes not part of `Check differences in broad_histology` chunk.
This will help us check what string assignment path_dx to short_histology mapping has changed from `molecular-subtyping-pathology`

```{r}

histology %>% 
  filter(!Kids_First_Biospecimen_ID %in% diff_broad_histology$Kids_First_Biospecimen_ID) %>%
   filter(toupper(short_histology.base) != toupper(short_histology.subtyped) 
         ) %>%
  select(Kids_First_Biospecimen_ID,starts_with("broad_histology"),starts_with("short_histology")) %>%
  unique() 

```
The above 172 changes occurred because of changes in string value assignment

> In `compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv` sample with broad_histology== `Ependymal tumor` is `EPN`, but in base histology it is `Ependymoma`.

Reviewer 2: ```suggestion
In `compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv` a sample with broad_histology that is `Ependymal tumor` is `EPN`, but in `pbta-histologies-base.tsv` it is `Ependymoma`.
```

> For samples with broad_histology==`Embryonal tumor`, short_histology is `Embryonal tumor` but in base histology it was `ETMR`.

Reviewer 2: ```suggestion
For samples where broad_histology is `Embryonal tumor`, short_histology is also `Embryonal tumor` but in base histology it was `ETMR`.
```

---
title: "Integrate molecular subtyping results"
output: 
  html_notebook:
    toc: true
    toc_float: true
author: Krutika Gaonkar for D3b
date: 2020
---

The purpose of this notebook is to integrate molecular subtyping results from 
[molecular-subtyping-pathology](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/molecular-subtyping-pathology) with `pbta-histologies-base.tsv`.

Here we will use `pbta-histologies-base.tsv` in which integrated_diagnosis,Notes and molecular_subtype are all NA. Through all the following molecular subtyping modules:

- molecular-subtyping-MB
- molecular-subtyping-CRANIO
- molecular-subtyping-EPN
- molecular-subtyping-embryonal
- molecular-subtyping-EWS
- molecular-subtyping-neurocytoma
- molecular-subtyping-HGG
- molecular-subtyping-LGAT
- molecular-subtyping-pathology

> we gathere and update molecular-subtype AND integrated_diagnosis AND broad_histology AND short_histology for these histologies.

Reviewer 2: ```suggestion
We gathered and updated molecular-subtype AND integrated_diagnosis AND broad_histology AND short_histology for these histologies.
```

> In this notebook we will add the molecular subtyping done as part as OpenPBTA to the base histology file to create the `pbta-histologies.tsv` for the same release. 

Reviewer 2: ```suggestion
In this notebook, we will add the molecular subtyping updated in the `pbta-histologies-base.tsv` file to update the `pbta-histologies.tsv` file. 
```

Author: > subtyping updated in the `pbta-histologies-base.tsv` 

This is not entirely true I mean `pbta-histologies-base.tsv` file is used but is not updated through the subtyping modules. The subtyping file is saved in 
compiled_subtyping<-read_tsv(file.path("..", "molecular-subtyping-pathology", "results", "compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv")) as part of the molecular-subtyping-pathology module.

We directly use compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv to update `pbta-histologies.tsv` file

Reviewer 2: Ah. Okay. Can you describe that here then?

Author: oh I forgot to commit the description 😅

Let me know if it makes sense, should I add in my workflow diagram if it's useful in the README or code or this looks ok ?

Reviewer 2: Sure! Why not add the diagram! Would probably be good to have that in this description.

Author: done!

- molecular-subtyping-MB
- molecular-subtyping-CRANIO
- molecular-subtyping-EPN
- molecular-subtyping-embryonal
- molecular-subtyping-EWS
- molecular-subtyping-neurocytoma
- molecular-subtyping-HGG
- molecular-subtyping-LGAT
- molecular-subtyping-pathology

we gathered and updated molecular-subtype/integrated_diagnosis/broad_histology/short_histology. 

In this notebook we will add the molecular subtyping done as part as OpenPBTA to the base histology file to create the `pbta-histologies.tsv` for the same release. 

For samples that where pathology_diagnosis is "Other" we also update the file broad_histology and short_histology from a manual review of WHO terms.

## Set up

```{r}
# This logic is the same as analyses/gene-set-enrichment-analysis/02-model-gsea.Rmd
# Assigning params$is_ci to running_in_ci avoids a locked binding error
running_in_ci <- params$is_ci

# Are we testing? In case of a non 0/1 number, we recast as logical, and then 
# ensure logical.
if (running_in_ci %in% c(0,1)) running_in_ci <- as.logical(running_in_ci)
if (!(is.logical(running_in_ci)))
{
  stop("\n\nERROR: The parameter `is_ci` should be FALSE/TRUE (or 0/1).")
}

library(tidyverse)

> if (running_in_ci) {
>   data_dir <- "../../data/"
> } else {
> data_dir <- "../../data/release-v18-20201123/"
> }

Reviewer 3: This conditional code should not be needed. The way the system is set up, all files should be linked to `data/` with no subfolder. In CI, this will be the testing files, but in regular use (after running `download-data.sh`) it will be the release version. Unless we want to make this file only work in v18 (I don't think we do) we should not be referring to the subfolder. 
```suggestion
data_dir <- "../../data/"
```

Reviewer: hmm - @jashapiro I think we did this because it was being run before CI files were generated. However, also thinking about the process, it is a bit circular. This needs to be run for `pbta-histologies.tsv` to be generated and added to the new release, before the CI files are generated. Thoughts on that?

Reviewer 3: The biggest concern I have is that somebody might rerun this script when generating v19+ and forget to update this line.  I don't know if this is pervasive through the subtyping files. 

(Of course, with the linking approach it is possible to have the wrong set of files linked, and get a hidden error that way, so neither one is great.)

Reviewer: @kgaonkar6 @jashapiro what if we add this as a requirement in the bash script in which the user has to put a release version?

Reviewer 3: I just looked at some of the other subtyping analyses, and we don't do that for those. I think for consistency with the other analyses, it should simply use `data/` and expect that the correct files have been placed or linked there.

Author: ok sounds good, I'll update the code to always use data
